packages/facility-server/__tests__/audit/attachChangelogToSnapshotRecords.test.js

```diff
@@ -0,0 +1,165 @@
+import { SYSTEM_USER_UUID } from "@tamanu/constants";
+import { attachChangelogToSnapshotRecords } from "@tamanu/database";
+import { createTestContext } from "../utilities";
+
+describe('attachChangelogToSnapshotRecords', () => {
+  let ctx;
+  let models;
+  beforeAll(async () => {
+    ctx = await createTestContext();
+    models = ctx.models;
+  });
+
+  afterAll(() => ctx.close());
+
+  afterEach(async () => {
+    const { ChangeLog } = models;
+    await ChangeLog.destroy({ where: {} });
+  });
+
+  it('should attach changelog records to snapshot records within the specified tick range', async () => {
+    const { ChangeLog } = models;
+    await Promise.all([
```

Comment:
This needs to have more negative cases covered, notably including changelog records for record id / table combos that aren't included in the snapshot records (I suspect this isn't catching the `or` that should be an `and`, if I'm right about that)